packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -130,6 +130,16 @@ class RunCommand extends RunCommandBase {
         defaultsTo: true,
         help: 'If necessary, build the app before running.',
       )
+      ..addOption('dart-flags',
+        hide: !verboseHelp,
+        help: 'Pass a list of comma separated flags to the Dart instance at '
+              'application startup. Flags passed through this option must be '
+              'present on the whitelist defined within the Flutter engine. If '
+              'a non-whitelisted flag is encountered, the process will be '
+              'terminated immediately.\n\n'
+              'This flag is only applied in debug and profile modes and is not '
```

Comment:
> This flag is only applied in debug and profile modes

This behavior needs a test.

Reply:
This is tested in the engine: https://github.com/flutter/engine/commit/12f48f760#diff-21bc42b7f14c143ec0333cc2e1212807R279

Reply:
The flutter_tool needs a test that the flags are not passed to the engine in a release run.

Reply:
The flags will be passed to the engine regardless where they're filtered out based on runtime mode. Not sure if that's just me misinterpreting the sematics here though.

Reply:
Below in `_createDebuggingOptions()`, the dart-flags are not included in the DebuggingOptions in a release build. That seems to me to be the correct behavior, and I think it would be good to make sure that behavior is retained by including a test with this change.

Reply:
Fair point. I'll add some tests.